hummingbot/market/bamboo_relay/bamboo_relay_order_book_tracker.py

```diff
@@ -215,18 +232,31 @@ def exchange_name(self) -> str:
                     message = await message_queue.get()
 
                 if message.type is OrderBookMessageType.DIFF:
-                    # bamboo relay does not support order book diffs
-                    pass
+                    # Diff message just refreshes the entire snapshot
+                    bids, asks = active_order_tracker.convert_diff_message_to_order_book_row(message)
+                    order_book.apply_snapshot(bids, asks, message.update_id)
+                    #past_diffs_window.append(message)
+                    #while len(past_diffs_window) > self.PAST_DIFF_WINDOW_SIZE:
+                    #    past_diffs_window.popleft()
+                    #diff_messages_accepted += 1
+
+                    # Output some statistics periodically.
+                    #now: float = time.time()
+                    #if int(now / 60.0) > int(last_message_timestamp / 60.0):
+                    #    self.logger().debug("Processed %d order book diffs for %s.",
+                    #                        diff_messages_accepted, trading_pair)
+                    #    diff_messages_accepted = 0
+                    #last_message_timestamp = now
                 elif message.type is OrderBookMessageType.SNAPSHOT:
-                    past_diffs: List[BambooRelayOrderBookMessage] = list(past_diffs_window)
+                    #past_diffs: List[BambooRelayOrderBookMessage] = list(past_diffs_window)
```

Comment:
did you accidentally leave in these commented out lines?

Reply:
if you added diff messages you might need to replay previous diff messages when processing a snapshot

Reply:
No, I changed the way the diff snapshot works so that it returns an entire snapshot to apply instead of a diff stream.
It's because the previous diff function (copied from radar) looks like it ends up erasing the entire book and only resulting in a single/bid ask on the book instead of applying the diff correctly.
So I changed it to just resort the bids/ask and return the entire list to apply.
I changed the WebSocket spec on my side so I send an array of updates (diffs) at a time now anyway.